### PR TITLE
[GEMINIEDA-455] Add "Free" option for placement

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
Add free option

 ### Which part of the code base require a change
 
 - [x] Library: compiler, newproject, projnavigator
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [ ] Regression tests
 - [ ] Continous Integration (CI) scripts

### Screenshot
image
image

### CODE REVIEW

File: src/Compiler/CompilerOpenFPGA.cpp
Comment (KochynVolodymyr) [inline src/Compiler/CompilerOpenFPGA.cpp:1604]: @alain-rs  please check if this is okay.

**Review by skyler-rs — APPROVED** [review]
Approved pending alain's review that volodymyr requested above

**Review by ravic-rs — APPROVED** [review]
Looks good. @alain-rs waiting for your approval on compiler changes.

File: src/Compiler/CompilerOpenFPGA.cpp
Comment (ravic-rs) [reply src/Compiler/CompilerOpenFPGA.cpp:1625]: @taoli4rs we need this change because it was not captured in the previous PR. Does this look ok to you ?

**Comment by ravic-rs** [comment]
Pending tests at upstream.

**Comment by ravic-rs** [comment]
> Pending tests at upstream.

 Verified with test case from Tao. Merging.

**Pull request merged**
